compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -727,6 +727,8 @@ object SymDenotations {
 
     /** Is this a denotation of a class that does not have - either direct or inherited -
      *  initialization code?
+     *
+     *  This method is intended to be used for classes only, not for traits.
```

Comment:
You could make that more explicit by calling it isNoInitsRealClass and call isRealClass instead of isClass

Reply:
Oh I didn't know there was a name for that. Will do.